### PR TITLE
Redesign agents stream UI: modes, filters, processors sheet

### PR DESCRIPTION
**Pull request by jonastemplestein**
## Summary / motivation

Agent stream pages were still oriented around a **Feed / State** split that no longer matches how people actually inspect agents. Operators want a clean chat transcript by default, a way to surface platform noise when debugging, and processor/runtime state without leaving the stream. This PR remodels the agent stream shell around **view modes** (React surfaces, not pure filters), a **mode-owned filter row**, and a proper **processors sheet**.

Non-agent domains keep the existing raw feed_items + domain presets behavior. Repos full-panel UI is unchanged (Events button path still works).

### Modes are components, not filters

- On `/agents/*`, the header shows **Pretty | Pretty + debug | Raw**.
- Modes choose which React surface mounts:
  - **Pretty** / **Pretty + debug** → `AgentFeedView` (agent-ui projection)
  - **Raw** → grouped `feed_items` + type/offset filters (same class of surface as non-agent streams)
- Non-agent streams do **not** show mode tabs; they stay on the raw feed with domain presets.
- Modes deep-link via `?mode=` and **replace** the old Feed/State `tab` mental model. Legacy `tab=feed|state` is still accepted so old links don't break; `state` no longer opens a State tab (processor state lives in the sheet).

### Pretty vs Pretty + debug

- Same `AgentFeedView` component for both.
- Pretty hides debug-only agent-ui kinds (`stream-woken`) in SQL.
- Pretty + debug shows all agent-ui kinds, including wake rows.
- Search (`q`) composes with the debug kind filter; dense LIMIT/OFFSET windows are used whenever the index space is filtered.

### Filters are mode-owned

- Filter icon only appears when the active mode opts in (`filters: true` on the mode definition).
- Icon toggles the filter row via `filter=true`; values stick in the URL when the row is closed.
- Pretty modes: search-only bar (+ event count / connection / mode label).
- Raw: presets + type chips + from/to offsets + search.
- Deep-linkable via TanStack search params.

### Header cleanup (agent pages)

- Dropped the **Running** badge.
- Dropped **Feed / State** tabs in favor of mode tabs.
- Presence avatars + RTT sparkline open the processors sheet (overview or focused processor).
- State is no longer a peer tab of the feed — it lives with processors.

### Processors sheet IA

- Proper shadcn **Sheet**, full height, ~55vw / max 48rem on desktop.
- **Overview**: sparkline metrics + presence list of consumers.
- **Detail**: avatar click focuses a processor (`processor=` subscription key).
- Agent processor reduced state has a **Pretty** renderer (phase, provider, model, failures, history teaser, system prompt, autonomous turns) with a **Raw** toggle for YAML/JSON.

## URL contract

| Param | Meaning |
| --- | --- |
| `mode` | `pretty` \| `pretty-debug` \| `raw` (omitted → default: pretty on agents, raw elsewhere) |
| `filter` | `true` when the mode's filter/search row is open |
| `q` | text search |
| `preset` | feed-items preset id (raw / non-agent) |
| `types` | event type any-of filters (raw) |
| `from` / `to` | inclusive offset bounds (raw) |
| `panel` | processors sheet open (overview when `processor` absent) |
| `processor` | subscription key focused in the sheet |
| `event` | raw-event inspector offset (mutually exclusive with processors sheet) |
| `events` | full-panel Events sheet (repos, etc.) |
| `tab` | **legacy** only (`feed` \| `state`); prefer `mode` |

Patches use `replace: true` so mode/filter clicks don't pile history entries.

## Screenshots

Captured against local dev (`http://localhost:5173`) on a real agent after one short chat turn.

### Pretty (default) — no wake rows

Pretty mode header + clean chat

Header: path pill, presence avatars, RTT sparkline, **Pretty | Pretty + debug | Raw**, filter icon. Feed is the agent-ui transcript without `stream-woken` noise.

### Pretty + debug — wakes + filter row

Pretty + debug with stream-woken and search

Same feed component; debug kinds (e.g. **Stream durable object woke**) appear. Filter icon opens search-only row (`filter=true`); values remain sticky when closed.

### Raw — grouped feed_items

Raw mode event list

Grouped raw feed with agent event types and offsets. Type/offset/preset filters available when the filter row is open.

### Processors sheet — overview

Processors sheet overview with RTT sparkline

Full-height sheet from presence/sparkline. Metrics + connected consumers.

### Processors sheet — agent pretty reduced state

Agent processor pretty reduced state

Detail view for the agent processor: offset/lag, phase/provider/model/failures, history teaser, system prompt, subscription key. **Raw** toggle for the serialized snapshot.

## Key files

- `apps/os/src/lib/stream-view-search.ts` — mode schema, `modesForStream`, panel helpers
- `apps/os/src/lib/stream-feed-filters.ts` (+ tests) — mode-aware filter activity
- `apps/os/src/components/stream-view-header.tsx` — mode tabs, filter toggle, presence → sheet
- `apps/os/src/components/stream-feed-filters.tsx` — mode-owned filter row
- `apps/os/src/components/agent-feed.tsx` — `showDebug` SQL filter for wakes
- `apps/os/src/components/stream-processors-panel.tsx` — Sheet + agent pretty state
- `apps/os/src/components/project-stream-view.tsx` — orchestrator wiring
- `apps/os/src/routes/.../integrations.tsx` — include `mode` in stream search reset

## Test plan

- [x] Local agent stream: Pretty default; Pretty + debug shows wake rows; Raw lists feed_items
- [x] `?mode=`, `?filter=true`, `?panel=true`, `?processor=` deep links
- [x] Filter icon toggles row; values stick when closed
- [x] Presence / sparkline open processors sheet; agent detail shows Pretty state + Raw toggle
- [x] Non-agent domain (`/agents` namespace stream, `/repos`): no mode tabs; raw presets still work
- [x] Repos full-panel: Events button still opens events sheet (no mode tabs in header)
- [x] Unit: `stream-feed-filters` tests for mode-aware active filters
- [ ] Smoke: switch streams via ⌘K with empty search resets mode/filters to defaults
- [x] Legacy `?tab=state` does not crash; lands on default mode (state via sheet)

## Review follow-ups (post-review commits)

- **Knip/dead code**: removed the orphaned `stream-state-view.tsx` and the dead type-filter dropdown exports; mode capability types are module-private now.
- **Bugbot — negative offsets**: `OffsetInput` clamps hand-typed negatives to 0 (offsets are non-negative), so `from`/`to` can no longer empty the raw feed.
- **Bugbot — mode clamping**: already fixed pre-review; `streamViewMode` falls back to the stream default when the URL asks for a mode the path doesn't offer.
- **React best-practice pass** (react-doctor: zero findings left in PR-touched files): `StreamModeTabs` extracted and shared by the header and the full-panel Events sheet; `ProjectStreamView` decomposed (`StreamEventsSheet`, `useAgentPauseControl`, `useStreamKillControl`); checkbox grids use `Set` lookups in render loops.
- Re-verified live on local dev after the refactor: mode tabs + full URL contract, Pretty hides `stream-woken` while Pretty+raw shows them, processors sheet overview/detail, repos Events sheet, legacy `?tab=state`.

## Non-goals / follow-ups

- **Mode registry abstraction** — `modesForStream` is still a small hard-coded table; a domain registry of view modes can come later.
- **More pretty renderers** — only the agent processor reduced state is pretty-rendered; other processors still use YAML/JSON.
- **Expand debug kinds** — only `stream-woken` is gated today; more debug-only agent-ui kinds can join the Pretty filter list.
- **Sheet scroll IA** — reduced state currently sits below owned-events contract docs; may want pretty state above the fold.
- PR screenshot assets under `docs/pr-assets/agents-stream-ui/` can be deleted after review if undesired in-tree.

---

> [!NOTE]
> **Medium Risk**
> Large UI refactor touching URL state, SQLite virtualized feeds, and operator controls (pause/kill); behavior is well-covered by new unit tests but regressions in scroll/filter edge cases are plausible.
> 
> **Overview**
> Replaces the **Feed / State** tab model with URL-backed **view modes** on agent streams: **Pretty** (chat only, hides `stream-woken` in SQL), **Pretty + raw** (chat + stacked raw rail, debug kinds visible), and **Raw** (`feed_items` only). Non-agent streams stay on an implicit raw feed with no mode tabs; invalid `?mode=` values clamp to each path’s default.
> 
> **`ProjectStreamView`** mounts surfaces from `modeCapabilities` instead of preset kind (`agent-chat` vs feed-items), always shows the composer (State tab removed), moves pause/kill into hooks, and extracts **`StreamEventsSheet`** for full-panel layouts. **`StreamModeTabs`** is shared by the header and Events sheet; mode switches clear feed-only URL params while keeping `q`.
> 
> **Filters** are mode-owned: Pretty gets search only; raw modes add presets, offset bounds, and an expandable **Types** panel with **`components`** (feed row shape) and **`types`** (event type), plus **`raw=false`** to hide the raw rail in Pretty+raw. **`buildFeedItemsFilter`** and **`feedItemsFilterFromSearch`** gained component filtering; the old event-type dropdown was removed from `feed-items-view`.
> 
> **Processors** open in a full-height **Sheet** (not an absolute overlay); agent processor reduced state gets a **Pretty** summary with a Raw toggle. **`stream-state-view.tsx`** is deleted—runtime state lives in the processors sheet. Header drops Running badge and Feed/State tabs; integrations route search reset includes `mode`, `components`, and `raw`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit eb32138f775f866a6d4463bc474b2bef8139b3c7. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

| Group | Lines | Significant |
| --- | ---: | ---: |
| Product | +301 -95 | +183 -19 🟩⬜⬜⬜⬜ |
| UI components | +999 -608 | +919 -532 🟩🟩🟩🟥🟥 |
| Tests | +114 -22 | +96 -13 🟩⬜⬜⬜⬜ |
| Docs | +0 -0 | +0 -0 ⬜⬜⬜⬜⬜ |
| **Total** | **+1,414 -725** | **+1,198 -564** 🟩🟩🟩🟥🟥 |

<sub>Lines counts every changed line; Significant ignores blank lines and JS comments. Between b3c5741 and eb32138, bucketed first-match-wins into the groups defined in `scripts/ci/loc-report.ts`.</sub>

## Environment Config Lease

<details>
<summary>No active environment config lease.</summary>

| app | status | commit | preview | size (gzip) | deploy duration | test duration | retries | cleanup duration | workflow run | updated | summary |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Auth | released | `eb32138` | [https://auth.iterate-preview-2.com](https://auth.iterate-preview-2.com) | 819.5 KiB | 26.3s | 589ms |  | 2.9s | [Workflow run](https://github.com/iterate/iterate/actions/runs/411565939310507) | 2026-07-09T23:17:44.132Z | Preview app released. |
| OS | released | `eb32138` | [https://os.iterate-preview-2.com](https://os.iterate-preview-2.com) | 4.14 MiB (-0.4 KiB vs main) | 67.5s | 206.9s |  | 11.0s | [Workflow run](https://github.com/iterate/iterate/actions/runs/411565939310507) | 2026-07-09T23:17:51.817Z | Preview app released. |
| Semaphore | released | `4b9ee52` | [https://semaphore.iterate-preview-2.com](https://semaphore.iterate-preview-2.com) | 440.8 KiB | 14.9s |  |  | 957ms | [Workflow run](https://github.com/iterate/iterate/actions/runs/92266433693084) | 2026-07-09T23:17:42.232Z | Preview app released. |
| Streams Example App | released | `eb32138` | [https://streams.iterate-preview-2.com](https://streams.iterate-preview-2.com) | 1.32 MiB | 14.6s | 18.6s |  | 970ms | [Workflow run](https://github.com/iterate/iterate/actions/runs/411565939310507) | 2026-07-09T23:17:42.251Z | Preview app released. |

</details>

### CODE REVIEW

File: apps/os/src/components/stream-feed-filters.tsx [hunk 1]
```diff
@@ -1,150 +1,450 @@
...
       onChange={(event) => {
-        const parsed = event.target.value === "" ? undefined : Number(event.target.value);
-        onChange(parsed != null && Number.isFinite(parsed) ? parsed : undefined);
+        const raw = event.target.value.trim();
+        if (raw === "") {
+          onChange(undefined);
+          return;
+        }
+        const parsed = Number(raw);
+        if (Number.isFinite(parsed)) onChange(Math.trunc(parsed));
```
[the remark this answers is not in the document]
Comment (jonastemplestein) [reply apps/os/src/components/stream-feed-filters.tsx]: Fixed in e0c2162e7 — `OffsetInput` now clamps to `Math.max(0, Math.trunc(parsed))`, so hand-typed negatives become 0 instead of flowing into `from`/`to`. Verified live: typing `-5` writes `from=0`.

File: apps/os/src/lib/stream-view-search.ts [hunk 2]
```diff
@@ -4,56 +4,99 @@ import { z } from "zod";
...
+
+export function defaultModeForStream(streamPath: string): StreamViewMode {
+  return streamPath.startsWith("/agents/") ? "pretty" : "raw";
+}
+
+export function streamViewMode(search: StreamViewSearch, streamPath: string): StreamViewMode {
+  if (search.mode != null) return search.mode;
+  // Stale `tab=state` links no longer open a State tab; fall through to default.
+  return defaultModeForStream(streamPath);
+}
```
[the remark this answers is not in the document]
Comment (jonastemplestein) [reply apps/os/src/lib/stream-view-search.ts:179]: Fixed in fa5b79e0f (before this review round): `streamViewMode` resolves the requested mode against `modesForStream(streamPath)` and falls back to the stream default when unsupported, so `pretty`/`pretty-raw` on non-agent paths can no longer leave hidden filters active. `feedFiltersActive` and the filter row both go through the same resolution.

File: apps/os/src/components/project-stream-view.tsx [hunk 3]
```diff
@@ -386,61 +365,11 @@ export function ProjectStreamView({
...
-            </div>
-            {filterRow}
-            {feedColumn}
-          </SheetContent>
-        </Sheet>
+        {processorsSheet}
+        <StreamEventsSheet streamPath={streamPath}>
+          {filterRow}
+          {feedColumn}
+        </StreamEventsSheet>
```
[the remark this answers is not in the document]
Comment (jonastemplestein) [reply apps/os/src/components/project-stream-view.tsx:372]: Fixed in 30287008a — overlay state resolves centrally in `useStreamViewPanels` (inspector beats processors, processors beats Events) and each opener clears the competing params, so `?events=true&panel=true` renders only the processors sheet. Verified live on the repos full-panel page.

**Comment by jonastemplestein** [comment]
## Preview e2e failure: fleet-wide, not this PR

The `Preview / deploy + e2e` failure is the onboarding smoke timing out (`saw 0 events`). Diagnosis from probes against the deployed preview-2 slot:

- Project creation, agent birth, and stream delivery all work — a probe project's `/agents/onboarding` stream fills with birth + LLM lifecycle events.
- **Every LLM turn fails with `2021: Invalid User Credentials` from the `env.AI` Workers AI binding for `openai/gpt-5.5`** (the #1828 default). The agent failure-loops and never emits the `agents/web-message-sent` greeting the smoke waits for.
- Not slot-specific and not this PR: #1824's preview failed identically at 23:06Z on its own slot (it also includes #1828), while the last green preview was #1828's own run at 22:55Z. This PR's diff is UI-only (stream view components).
- Secondary observation: the smoke's `waitForEvent` error says `saw 0 events; recent types: none` even though the watched stream has 30+ events — worth a look at the waiter's delivery lane messaging while fixing the above.

Looks like the preview Cloudflare account's partner-model (OpenAI BYOK / unified billing) credentials broke around 23:00Z on 2026-07-09. Every preview run fleet-wide will stay red until that's fixed or the default model is changed.

Merging on green required checks (Lint+Typecheck, Test) + clean Bugbot; the preview failure is environmental.
